ihp-sg13g2/libs.tech/klayout/python/sg13g2_pycell_lib/ihp/via_stack_code.py

```diff
@@ -0,0 +1,186 @@
+        
+        #*************************************************************************
+        #*
+        #* Main body of code
+        #*
+        #************************************************************************
+
+        for layer in metal_layers[metal_layers.index(b_layer):metal_layers.index(t_layer)+1]:
+
+            #pre-procesing
+            if layer == 'Metal1':
+                columns = vn_columns
+                rows = vn_rows
+                via_size = v1_size
+                via_sep = v1_sep1 if (columns<4 and rows<4) else v1_sep2
+                via_enc = v1_enc
+                w_x = (columns * via_size + (columns - 1) * via_sep)
+                w_y = (rows * via_size + (rows - 1) * via_sep)
+
+            elif layer == 'TopMetal1':
+                via_size = Topvia1_size
+                via_sep = TopVia1_sep
+                via_enc = Topvia1_enc_met5
+                columns = vt1_columns
+                rows = vt1_rows
+                w_x = (columns * via_size + (columns - 1) * via_sep)
+                w_y = (rows * via_size + (rows - 1) * via_sep)
+                dbCreateRect(self, 'Metal5', Box(-via_enc-w_x/2, -via_enc-w_y/2, w_x/2 + via_enc, w_y/2 + via_enc))
+                via_enc = Topvia1_enc_top1
+
+            elif layer == 'TopMetal2':
+                via_size = Topvia2_size
+                via_sep = TopVia2_sep
+                via_enc = Topvia2_enc_top1
+                columns = vt2_columns
+                rows = vt2_rows
+                w_x = (columns * via_size + (columns - 1) * via_sep)
+                w_y = (rows * via_size + (rows - 1) * via_sep)
+                dbCreateRect(self, 'TopMetal1', Box(-via_enc-w_x/2, -via_enc-w_y/2, w_x/2+via_enc, w_y/2+via_enc))
+                via_enc = Topvia2_enc_top2
+
+            else:
+                columns = vn_columns
+                rows = vn_rows
+                via_size = vn_size
+                via_sep = vn_sep1 if (columns<4 and rows<4) else vn_sep2
+                via_enc = vn_enc
+                w_x = (columns * via_size + (columns - 1) * via_sep)
+                w_y = (rows * via_size + (rows - 1) * via_sep)
+
+            #metal draw
+            if layer == 'TopMetal1':
+                if columns * via_size + (columns - 1) * 2 * via_sep < TopMetal1_min:
+                    via_enc = (TopMetal1_min - Topvia1_size)/2
+                dbCreateRect(self, layer, Box(-via_enc-w_x/2, -via_enc-w_y/2, w_x/2 + via_enc, w_y/2 + via_enc))
+            elif layer == 'TopMetal2':
+                if columns * via_size + (columns - 1) * 2 * via_sep < TopMetal2_min:
+                    via_enc = (TopMetal2_min - Topvia2_size) / 2
+                dbCreateRect(self, layer, Box(-via_enc-w_x/2, -via_enc-w_y/2, w_x/2 + via_enc, w_y/2 + via_enc))
+            else:
+                dbCreateRect(self, layer, Box(-via_enc-w_x/2, -via_enc-w_y/2, w_x/2 + via_enc, w_y/2 + via_enc))
```

Comment:
@lild4d4, what do you think about putting `dbCreateRect` outside of the conditional block?

Reply:
Yes !, It is a good idea. I didn't see that possibility when I was coding.